rstudio/rstudio
# Log to file for Electron desktop

### Intent

Implement desktop logging to file `rdesktop.log` in same location as qt desktop. #10444

### Approach

Use the `winston` package for logging. Also, if stdout is detected to be a TTY, also log to stdout (handy when running from command-line / yarn start).

Note that default logging level is 'warn'. To run with most verbose logging level, use `--log-level=Debug`.

To do this from `yarn start` you must do this dance:

`yarn start -- -- --log-level=Debug`

### Automated Tests

Removed some now-obsolete tests. Tweaked others to handle the changes.

### QA Notes

Try running with (for example) `/Applications/RStudio.app/Contents/MacOS/RStudio --log-level=Debug` to ensure stuff gets logged. Then use Help / Diagnostics / Show Log Files and look at `rdesktop.log`.

I'm using the default `winston` format (json) for now. We can adjust this later if we find a reason.

Additionally, not rotating the logs, so the rdesktop.log will grow. Winston has ability to rotate log files based on various criteria (https://www.npmjs.com/package/winston-daily-rotate-file); will open an issue to track implementing that. #10478

### Checklist
